network/websocket.ksy

```diff
@@ -0,0 +1,77 @@
+meta:
+  id: websocket_dataframe
+  title: WebSocket
+  xref:
+    rfc: 6455
+  endian: be
+doc: |
+  The WebSocket Protocol enables two-way communication between a client
+  running untrusted code in a controlled environment to a remote host
+  that has opted-in to communications from that code.  The security
+  model used for this is the origin-based security model commonly used
+  by web browsers.  The protocol consists of an opening handshake
+  followed by basic message framing, layered over TCP.  The goal of
+  this technology is to provide a mechanism for browser-based
+  applications that need two-way communication with servers that does
+  not rely on opening multiple HTTP connections (e.g., using
+  XMLHttpRequest or <iframe>s and long polling).
+seq:
+  - id: b1
+    type: u1
+  - id: b2
+    type: u1
+  - id: extended_len_1
+    type: u2
+    if: payload_len == 126
+  - id: extended_len_2
+    type: u4
+    if: payload_len == 127
+  - id: mask_key
+    type: u4
+    if: mask == 1
+  - id: payload
+    size: len
+    
+instances:
+  fin:
+    value: (b1 & 0b10000000) >> 7
+  rsv1:
+    value: (b1 & 0b01000000) >> 6
+  rsv2:
+    value: (b1 & 0b00100000) >> 5
+  rsv3:
+    value: (b1 & 0b00010000) >> 4
```

Comment:
You can use bit-level types (i.e. `type: b1`) instead of doing all these arithmetics.

Reply:
Done! I was unable to apply the bit types to the mask/payload values given the need for comparison in `mask_key`.